lib/puppet/util/feature.rb

```diff
@@ -3,40 +3,52 @@
 class Puppet::Util::Feature
   attr_reader :path
 
-  # Create a new feature test.  You have to pass the feature name,
-  # and it must be unique.  You can either provide a block that
-  # will get executed immediately to determine if the feature
-  # is present, or you can pass an option to determine it.
-  # Currently, the only supported option is 'libs' (must be
-  # passed as a symbol), which will make sure that each lib loads
-  # successfully.
-  def add(name, options = {})
+  # Create a new feature test. You have to pass the feature name, and it must be
+  # unique. You can pass a block to determine if the feature is present:
+  #
+  #     Puppet.features.add(:myfeature) do
+  #       # return true or false if feature is available
+  #     end
+  #
+  # The block should return true if the feature is available, false if it is
+  # not, or nil if the state is unknown. True and false values will be cached. A
+  # nil value will not be cached, and should be used if the feature may become
+  # true in the future.
+  #
+  # Feature are often used to detect if a ruby library is installed. To support
```

Comment:
`s/Feature/Features/`?

Also might be worth mentioning in the updated docs that the options hash is ignored if a block is given.

Reply:
I updated the comment for both things in https://github.com/puppetlabs/puppet/pull/7008/commits/77614383f81728dfa7bf329a41695780555b17fc#diff-f6dce0fbcf55caef42fe28d46c33e340R13

Reply:
This bit?
> Options passed to the block are ignored.

I was thinking  something more in line with your commit message in a8de657d602ac526991bc0d02828f4093e960701:
> If a feature is defined using `:libs` and a block, then the block is used and the `:libs` are ignored.

Reply:
Ah k, should be fixed now in https://github.com/puppetlabs/puppet/pull/7008/commits/e59bb6441640cfa4a9a7d8e4e1bc40c9bfedf87b#diff-f6dce0fbcf55caef42fe28d46c33e340R6